reascripts/ReaSpeech/source/ReaSpeechWorker.lua

```diff
@@ -245,11 +248,19 @@ function ReaSpeechWorker:check_active_job()
     self:check_active_job_request_status()
   end
 
-  if active_job.transcript_request then
-    self:check_active_job_transcript_request_status()
-  else
-    self:check_active_job_status()
+  if #active_job.requests > 0 then
+    for key, request in pairs(active_job.requests) do
+      if type(request) ~= 'function' then
+        if not request:ready() then return end
+      else
+        active_job.requests[key] = request()
+        return
+      end
+    end
+    self:handle_responses()
```

Comment:
the looping/return logic here is mostly about handling one request at a time. if it gets through the loop all the way (all requests are `ready`), then it can proceed to bundle the responses.

need to do some testing here for a possible logic error re: `error()` state